### DISCUSSION TITLE
Add generic types for prompt variables in the sdk clients, e.g. `langfuse.getPrompt<{input: string}>("prompt name")`

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#3817] [Ideas]
### Describe the feature or potential improvement

User request
> While we are talking about typecsript and promptclient, it would be cool to add some typesafety for the prompt variables (or config). Getting them generated is not an option of course, but maybe you could do smth like what postgres.js is doing here so I could `ChatPromptClient<{config: {a: number}, variables: {movie: string}}>` and then the `.compile()` and `.config` would be typed. It's not real type safety of course, but it's much better than nothing! This is a minor issue though.

Useful additions:
- Copy-pastable code in the Langfuse UI that already includes these types

**Comment by usersina** [comment]
I just created a type that `extends Record<string, string>` and provide the variable of that type to the `ChatPromptClient.compile`.
That said, I would love to see an SDK generator tool using our existing prompts since it's getting tough to manage...
